tests/codegen/paginator-tests/jvm/src/DefaultLambdaClient.kt

```diff
@@ -0,0 +1,41 @@
+package com.test
+
+import com.test.model.FunctionConfiguration
+import com.test.model.ListFunctionsRequest
+import com.test.model.ListFunctionsResponse
+
+/**
+ * This is a stub that produces pages of responses.  The marker is a series of dots "...", where
+ * each dot represents a service round trip.  Within each service round trip a list of [FunctionConfiguration]
+ * is generated including the round trip index.
+ */
+class DefaultLambdaClient(config: LambdaClient.Config) : LambdaClient {
```

Comment:
> Won't this conflict with the generated DefaultLambdaClient? Or is it not generating one because no protocol generator is found?

Yes.  In fact the test client must be called this due to reference to the type name in `LambdaClient`.

Reply:
`LambdaClient.Config` is also required for compilation to succeed

Reply:
I see, can we document what is happening here because it was confusing to me at first glance. Something to the effect of `The default client is not generated because no protocol generator is found/available, we supply a hand written version instead suitable for these tests`


Not something in this PR but it seems like we should maybe make the generated clients not depend on `DefaultLambdaClient`. IDK I mean I guess in practice it should normally always be there.

Reply:
Added a note to describe the situation as requested.

> seems like we should maybe make the generated clients not depend on DefaultLambdaClient.

Yeah I think this is the rough edge of where a proper Smithy SDK would be designed and built.

Reply:
Verified that I can address these concerns after https://github.com/awslabs/smithy-kotlin/pull/564 is merged.